src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataJsonReader.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading.Tasks;
+using System.Text.Json;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+
+namespace Microsoft.Azure.Functions.Worker
+{ 
+    internal class FunctionMetadataJsonReader
```

Comment:
Should we consider renaming this type? This class is more of a Function load response generator, than a JSON reader.

Reply:
Also, the method prefix could be `Get` (Ex: `GetFunctionLoadResponse` or so).

Reply:
The naming convention was kept from a similar class that read from the `functions.metadata` file. It does not generate any function metadata but "reads" it from a file.

Shyju and I discussed this in a call, and we are deciding between `FunctionMetadataProvider` or `FunctionMetadataReader`, and depending on how this class will be used in later iterations, whether or not to keep "Json" in the name.